src/hotspot/share/c1/c1_ValueMap.cpp

```diff
@@ -585,7 +589,7 @@ void GlobalValueNumbering::substitute(Instruction* instr) {
   if (subst != instr) {
     assert(!subst->has_subst(), "can't have a substitution");
 
-    TRACE_VALUE_NUMBERING(tty->print_cr("substitution for %d set to %d", instr->id(), subst->id()));
+    TRACE_VALUE_NUMBERING(tty->print_cr("substitution for %c%d set to %c%d", instr->type()->tchar(), instr->id(), subst->id(), subst->type()->tchar()));
```

Comment:
Shouldn't the third argument be a char and the fourth an int?

Reply:
Good catch, fixed.